source/development/coredump.rst

```diff
@@ -0,0 +1,160 @@
+
+#. **Disable Core Dump Generation**:
+    To disable core dump generation, set the core dump size limit to zero using the ``ulimit`` command:
+
+    .. code-block:: console
+
+        # ulimit -c 0
+
+#. **Optional: Configure Core Dump Storage Location**:
+    If core dumps were previously being stored, you may want to configure the ``core_pattern`` sysctl parameter to prevent any future core dumps from being generated. For example, you can set it to ``/dev/null`` to discard core dumps:
+
+    .. code-block:: console
+
+        # echo "/dev/null" > /proc/sys/kernel/core_pattern
+
+#. **Restart Processes if Necessary**:
+    If you've changed the ``core_pattern`` parameter, consider restarting relevant processes to ensure that the changes take effect.
+
+By following these steps, you can disable core dump generation without relying on systemd. This approach directly modifies system-wide settings to prevent core dumps from being generated.
+
+MacOS agent's OS
+----------------
+On macOS, core dump generation is disabled by default for most applications. However, you can enable core dump generation for specific processes using the ``ulimit`` command. Here's how you can enable core dump generation on macOS:
+
+#. **Check Current Core Dump Configuration**:
+    Before enabling core dump generation, check the current core dump size limit using the ``ulimit`` command:
+
+    .. code-block:: console
+
+        # ulimit -c
+        # sysctl kern.corefile
+
+#. **Identify the Target Directory for Core Dumps**:
+    On macOS, core dump files are typically stored in the current working directory of the process that crashes.
+
+#. **Enable Core Dump Generation**:
+    To enable core dump generation for a specific process, set the core dump size limit to a non-zero value using the ``ulimit`` command. For example, to set the limit to unlimited:
+
+    .. code-block:: console
+
+        # ulimit -c unlimited
+        # sysctl -w kern.corefile=/cores/core.%P
+
+By following these steps, you can enable core dump generation for specific processes on macOS. Keep in mind that enabling core dump generation may consume additional disk space, so use it judiciously. Additionally, core dump generation may not be supported or may behave differently for all processes on macOS.
+
+To disable coredump generation, you can ensure that core dumps are not generated by setting the core dump size limit to zero. Here's how you can disable core dump generation on macOS:
+
+    .. code-block:: console
+
+        # ulimit -c 0
+
+By setting the core dump size limit to zero, you ensure that core dumps are not generated for any processes on macOS. Keep in mind that this setting affects the entire system and may impact troubleshooting capabilities in case of application crashes.
```

Comment:
Is recommending to set the limit to 0 a good recommendation then? It seems something to avoid in my view from reading this. Is `ulimit -c` a general command or specific to certain processes?

Reply:
The `ulimit` (user limit) command is used to configure some other limits, in the case of `coredump` (-c) it is correct to set the limit to 0 (disable it).
I'll delete the last comment to avoid confusion.